metrics/src/metadata.rs

```diff
@@ -0,0 +1,65 @@
+/// Describes the level of verbosity of a metric event.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Level(LevelInner);
+
+impl Level {
+    /// The "error" level.
+    pub const TRACE: Self = Self(LevelInner::Trace);
+    /// The "warn" level.
+    pub const DEBUG: Self = Self(LevelInner::Debug);
+    /// The "info" level.
+    pub const INFO: Self = Self(LevelInner::Info);
+    /// The "debug" level.
+    pub const WARN: Self = Self(LevelInner::Warn);
+    /// The "trace" level.
+    pub const ERROR: Self = Self(LevelInner::Error);
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+enum LevelInner {
+    Trace = 0,
+    Debug = 1,
+    Info = 2,
+    Warn = 3,
+    Error = 4,
+}
+
+/// Metadata describing a metric event. This provides additional context to [`Recorder`](crate::Recorder), allowing for
+/// fine-grained filtering.
+/// 
+/// Contains the following:
+/// 
+/// - A [`target`](Metadata::target), specifying the part of the system where the metric event occurred. When
+/// initialized via the [metrics macro](metrics_macro), and left unspecified, this defaults to the module path the
```

Comment:
Looks like this needs to be changed to something like `[metrics macros][metrics_macros]` and then add `[metrics_macros]: https://docs.rs/metrics/latest/metrics/#macros` at the bottom of the overall doc comment.

I don't think there's a way to do intradoc links with _just_ a crate name, but it's also cleaner/more intuitive, I think, to link the macros section of the `metrics` crate since that's where people will be consuming them from.